### PR TITLE
Found a fix for rollback

### PR DESCRIPTION
**Pull request by AshciR**

### CODE REVIEW

File: src/shout_subgroup/database.py [hunk 1]
```diff
@@ -33,13 +33,12 @@ def configure_database() -> bool:
         logger.exception(f"Unable to connect to postgreSQL. See exception details ... {ex}")
         return False
 
-    global session
+    global Session
 
-    session_maker = sessionmaker(bind=engine)
-    session = session_maker()
+    Session = sessionmaker(bind=engine, expire_on_commit=False)
```
Comment (AshciR) [inline src/shout_subgroup/database.py:38]: `sessionmaker` creates a factory which we use to start db sessions. But the session isn't started until the function is called.

> a session establishes all conversations with the database and represents a “holding zone” for all the objects which you’ve loaded or associated with it during its lifespan

The mistake we made before was that we started a session, but we never closed it (manually or automatically).

File: src/shout_subgroup/group_chat_listener.py [hunk 2]
```diff
@@ -88,41 +88,46 @@ async def listen_for_messages_handler(update: Update, context: ContextTypes.DEFA
...
-    try:
-        maybe_added_user = await add_user_to_group_chat(session, update.effective_chat, user_who_sent_the_message)
-        # TODO: Add message like "The bot has recognized John Doe"
-        return
-    except NotGroupChatError:
-        # await update.message.reply_text("Sorry, you can only create or modify subgroups in group chats.")
-        return
+    db_session = get_database()
+
+    with db_session.begin() as session:
```
Comment (AshciR) [inline src/shout_subgroup/group_chat_listener.py:93]: We get the session factory from `get_database()` then use `.begin()` to start the session. When we use `.begin()` within the context of a `with` block. We get the `rollback()` and `commit()` handling for free.

https://docs.sqlalchemy.org/en/20/orm/session_basics.html#using-a-sessionmaker

File: src/shout_subgroup/repository.py [hunk 3]
```diff
@@ -137,7 +137,7 @@ async def insert_user(
         last_name=last_name
     )
     db.add(new_user)
-    db.commit()
+    db.flush()
```
Comment (AshciR) [inline src/shout_subgroup/repository.py:140]: B/c we're handling the `.commit()` implicitly within the `with` block, calling it here prematurely stopped the session and the subsequent db calls failed. 

Also, we have to use `.flush()` when after .add(). TBH, I don't fully understand why.
https://docs.sqlalchemy.org/en/20/orm/session_basics.html#session-flushing

File: src/shout_subgroup/group_chat_listener.py (hunk 2 above)
Comment (AshciR) [inline src/shout_subgroup/group_chat_listener.py:93]: Also, I'm using a pattern I found in [FastAPI documentation](https://fastapi.tiangolo.com/tutorial/sql-databases/?h=data#create-a-dependency). The pattern is to create a db session for each API request (command handlers in our case), complete the transaction, then close the db session

**Review by rick1e — APPROVED** [review]

**Pull request merged**